src/GCPStorageService.js

```diff
@@ -299,14 +299,20 @@ export class GCPStorageService extends BaseStorageService {
    * @description                     - Generates a downloadable URL for a file in the gcloud bucket.
    * @param {string} container        - gcloud bucket name.
    * @param {string} filePath         - Path to the file in the bucket.
+   * @param {number} expiresIn        - Expiry time for the signed URL in seconds. Default is 3600.
+   * @param {string} permission       - Permission for the operation. Use WRITE for PUT operations.
    * @returns {Promise<string>}       - A downloadable URL for the specified file.
    */
-  async getDownloadableUrl(container, filePath) {
+  async getDownloadableUrl(container, filePath,expiresIn = 3600, permission = '') {
```

Comment:
Keep the same function signature as here - https://github.com/Sunbird-Ed/client-cloud-services/blob/ecd1e99dcf22ee06fc151fe2697ac352a2a3e889/src/AzureStorageService.js#L409 No need of permission.

Reply:
But for public bucket without calling getSignedUrl  we can get downloadbleURl @aks30 , Only for the private we need to call getSignedUrl , that's why used permission "r"

Reply:
Then let's add similar description on line 303, when this value should be passed and what should be the value which is allowed.

Reply:
Sure @aks30  will add that